src/gui/src/ActionDialog.cpp

```diff
@@ -113,5 +103,14 @@ void ActionDialog::keySequenceChanged()
     }
 }
 
+void ActionDialog::actionTypeChanged(int index)
+{
+    ui_->keySequenceWidget->setEnabled(index < 3);
```

Comment:
It's very unclear what each index value refers to, as the definition is in the UI. I think better idea would be to define combobox options in the source code:

```
enum ActionDialogActionType {
    ACTION_TYPE_ABC = 0,
    ACTION_TYPE_XYZ,
    ...
}

...

// keep in sync with ActionDialogActionType
std::vector<QString> s_action_type_names = { "abc", "xyz", ... };

...

// in constructor

for (const auto& name : s_action_type_names) {
    ui_->combobox->addItem(name);
}

...

// in actionTypeChanged

ui_->keySequenceWidget->setEnabled(index == ACTION_TYPE_XYZ || index == ACTION_TYPE_ABC ... );

```